src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+            bytes_total: base_idx.try_into().unwrap_or(0u128),
+        })
+    }
+}
+
+impl Output<File> {
+    fn write_blocks(&mut self, buf: Vec<u8>) -> io::Result<WriteStat> {
+        let mut writes_complete = 0;
+        let mut writes_partial = 0;
+        let mut base_idx = 0;
+
+        while base_idx < buf.len() {
+            let next_blk = cmp::min(base_idx + self.obs, buf.len());
+            let plen = next_blk - base_idx;
+
+            match self.write(&buf[base_idx..next_blk])? {
+                wlen if wlen < plen => {
+                    writes_partial += 1;
+                    base_idx += wlen;
+                }
+                wlen => {
+                    writes_complete += 1;
+                    base_idx += wlen;
+                }
+            }
+        }
+
+        Ok(WriteStat {
+            writes_complete,
+            writes_partial,
+            bytes_total: base_idx.try_into().unwrap_or(0u128),
+        })
+    }
+}
+
+/// Splits the content of buf into cbs-length blocks
+/// Appends padding as specified by conv=block and cbs=N
+fn block(buf: Vec<u8>, cbs: usize, rstat: &mut ReadStat) -> Vec<Vec<u8>> {
+    let mut blocks = buf
+        .split(|&e| e == b'\n')
+        .fold(Vec::new(), |mut blocks, split| {
+            let mut split = split.to_vec();
+            if split.len() > cbs {
+                rstat.records_truncated += 1;
+            }
+            split.resize(cbs, b' ');
+            blocks.push(split);
+
+            blocks
+        });
+
+    if let Some(last) = blocks.last() {
+        if last.iter().all(|&e| e == b' ') {
+            blocks.pop();
+        }
+    }
+
+    blocks
+}
+
+/// Trims padding from each cbs-length partition of buf
+/// as specified by conv=unblock and cbs=N
+fn unblock(buf: Vec<u8>, cbs: usize) -> Vec<u8> {
+    // Local Helper Fns ----------------------------------------------------
+    #[inline]
+    fn build_blocks(buf: Vec<u8>, cbs: usize) -> Vec<Vec<u8>> {
+        let mut blocks = Vec::new();
+        let mut curr = buf;
+        let mut next;
+        let mut width;
+
+        while !curr.is_empty() {
+            width = cmp::min(cbs, curr.len());
+            next = curr.split_off(width);
+
+            blocks.push(curr);
+
+            curr = next;
+        }
+
+        blocks
+    }
+    // ---------------------------------------------------------------------
+    build_blocks(buf, cbs)
+        .into_iter()
+        .fold(Vec::new(), |mut unblocks, mut block| {
+            let block = if let Some(last_char_idx) = block.iter().rposition(|&e| e != b' ') {
+                block.truncate(last_char_idx + 1);
+                block.push(b'\n');
+
+                block
+            } else if let Some(b' ') = block.get(0) {
+                vec![b'\n']
+            } else {
+                block
+            };
+
+            unblocks.push(block);
+
+            unblocks
+        })
+        .into_iter()
+        .flatten()
+        .collect()
```

Comment:
```suggestion
    buf.as_chuncks(cbs)
        .map(|mut block| {
            if let Some(last_char_idx) = block.iter().rposition(|&e| e != b' ') {
                block.truncate(last_char_idx + 1);
                block.push(b'\n');

                block
            } else if let Some(b' ') = block.get(0) {
                vec![b'\n']
            } else {
                block
            }
        })
        .flatten()
        .collect()
```

Reply:
Good spot on this one! I ended up removing a large amount of cruft by incorporating your feedback on this and [the previous](https://github.com/uutils/coreutils/pull/2474#discussion_r668709056) suggestion.